packages/mdc-menu/simple/foundation.js

```diff
@@ -399,69 +335,202 @@ class MDCSimpleMenuFoundation extends MDCFoundation {
+    if (bottomOverflow > 0 && topOverflow < bottomOverflow) {
+      corner |= CornerBit.BOTTOM;
+    }
 
-    const topOverflow = anchor.top + this.dimensions_.height - windowDimensions.height;
-    const bottomOverflow = this.dimensions_.height - anchor.bottom;
-    const extendsBeyondTopBounds = topOverflow > 0;
+    const leftAligned = this.adapter_.isRtl() ? (this.anchorCorner_ & (CornerBit.FLIP_RTL | CornerBit.RIGHT)) :
+      !(this.anchorCorner_ & CornerBit.RIGHT);
+    const availableLeft = leftAligned ? viewportMargin.left + this.anchorMargin_.left :
+      viewportMargin.left + anchorWidth + this.anchorMargin_.right;
+    const availableRight = leftAligned ? anchorWidth - this.anchorMargin_.left + viewportMargin.right :
+      viewportMargin.right - this.anchorMargin_.right;
 
-    if (extendsBeyondTopBounds) {
-      if (bottomOverflow < topOverflow) {
-        vertical = 'bottom';
+    const leftOverflow = menuWidth - availableLeft;
+    const rightOverflow = menuWidth - availableRight;
+    if (this.adapter_.isRtl()) {
+      if (leftOverflow < 0 && this.anchorCorner_ & CornerBit.FLIP_RTL & ~CornerBit.RIGHT) {
+        corner |= CornerBit.RIGHT;
       }
+    } else if (rightOverflow > 0 && leftOverflow < rightOverflow) {
+      corner |= CornerBit.RIGHT;
     }
 
-    const leftOverflow = anchor.left + this.dimensions_.width - windowDimensions.width;
-    const rightOverflow = this.dimensions_.width - anchor.right;
-    const extendsBeyondLeftBounds = leftOverflow > 0;
-    const extendsBeyondRightBounds = rightOverflow > 0;
+    return corner;
+  }
 
-    if (this.adapter_.isRtl()) {
-      // In RTL, we prefer to open from the right.
-      horizontal = 'right';
-      if (extendsBeyondRightBounds && leftOverflow < rightOverflow) {
-        horizontal = 'left';
+  /**
+   * @param {Corner} corner Origin corner of the menu.
+   * @return {number} Horizontal offset of menu origin corner from corresponding anchor corner.
+   * @private
+   */
+  getHorizontalOriginOffset_(corner) {
+    const {anchorRect} = this.measures_;
+    const horizontalAlignment = (corner & CornerBit.RIGHT) ? 'right' : 'left';
+    let x = 0;
+    if (horizontalAlignment === 'right') {
+      const rightOffset = (this.anchorCorner_ & CornerBit.RIGHT) ?
+        anchorRect.width - this.anchorMargin_.left : this.anchorMargin_.right;
+      x = rightOffset;
+    } else {
+      const leftOffset = (this.anchorCorner_ & CornerBit.RIGHT) ?
+        anchorRect.width - this.anchorMargin_.right : this.anchorMargin_.left;
+      x = leftOffset;
+    }
+    return x;
+  }
+
+  /**
+   * @param {Corner} corner Origin corner of the menu.
+   * @return {number} Vertical offset of menu origin corner from corresponding anchor corner.
+   * @private
+   */
+  getVerticalOriginOffset_(corner) {
+    const {anchorRect, viewport, viewportMargin, anchorHeight, menuHeight} = this.measures_;
+    const verticalAlignment = (corner & CornerBit.BOTTOM) ? 'bottom' : 'top';
```

Comment:
I'm not sure there's really a point to this variable here, could just inline `corner & CornerBit.BOTTOM` below in the `if`?

Reply:
I think it is a little bit more readable that way.

Reply:
If you want a separate variable, I'd go for something like `isBottomAligned` again instead, that way you can simply do `if (isBottomAligned)` rather than comparing to arbitrary strings. Same for the next instance below.